src/main/webapp/map.js

```diff
@@ -120,14 +144,54 @@ function createMarker(place) {
 }
 
 function displayMarkers() {
-  for (var i = 0; i < markers.length; i++ ) {
-    markers[i].setMap(map);
-  }
+  markers.forEach(marker => marker.setMap(map));
 }
 
 function clearMarkers() {
-  for (var i = 0; i < markers.length; i++ ) {
-    markers[i].setMap(null);
-  }
+  markers.forEach(marker => marker.setMap(null));
   markers = [];
 }
+
+/**
+/* Calculates the straight line distance between two
+/* sets of latitutes and longitutes
+/* A breakdown of the haversine formula can be found at
+/* http://www.movable-type.co.uk/scripts/latlong.html
+ */
+function haversine_distance(lat1,lon1,lat2,lon2) {
+  var R = 6371; // Radius of the earth in km
+  var dLat = degToRad(lat2-lat1);
+  var dLon = degToRad(lon2-lon1); 
+  var a = 
+    Math.sin(dLat/2) * Math.sin(dLat/2) +
+    Math.cos(degToRad(lat1)) * Math.cos(degToRad(lat2)) * 
+    Math.sin(dLon/2) * Math.sin(dLon/2);
+  var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a)); 
+  var d = R * c; // Distance in km
+  return d;
+}
+
+function degToRad(deg) {
+  return deg * (Math.PI/180)
+}
+
+/**
+/* Gets the location and name of the place
+/* and displays the walking route on Google Map
```

Comment:
Will we display the walking route to the TYD Coffee user?

Reply:
I don't think we are anymore. We still need the API call to get estimate time, but I will delete the call to DirectionsRenderer.